### PR TITLE
bind wlr_input_method_v2.h

### PR DESCRIPTION
**Pull request by novakne**

### CODE REVIEW

**Comment by novakne** [comment]
Fixed some errors based on the other PR

File: src/types/input_method_v2.zig [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    pub const DeleteSurroundingText = extern struct {
+        before_length: u32,
+        after_length: u32,
+    };
+
+    pub const State = extern struct {
+        preedit: wlr.InputMethodV2.PreeditString,
+        commit_text: [*:0]u8,
+        delete: wlr.ImputMethodV2.DeleteSurroundingText,
```
Comment (ifreund) [inline src/types/input_method_v2.zig:38]:
```suggestion
        delete: wlr.InputMethodV2.DeleteSurroundingText,
```

File: src/types/input_method_v2.zig [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    };
+
+    pub const KeyboardGrab = extern struct {
+        ressource: *wl.Ressource,
```
Comment (ifreund) [inline src/types/input_method_v2.zig:42]:
```suggestion
        resource: *wl.Resource,
```

File: src/types/input_method_v2.zig [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        input_method: *wlr.InputMethodV2,
+        keyboard: *wlr.Keyboard,
```
Comment (ifreund) [inline src/types/input_method_v2.zig:44]:
```suggestion
        keyboard: ?*wlr.Keyboard,
```

File: src/types/input_method_v2.zig [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            destroy: wl.Signal(*wlr.InputMethodV2.KeyboardGrab),
+        },
+
+        extern fn wlr_input_method_keyboard_grab_v2_send_key(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, time: u32, key: u32, state: u32) void;
+        pub const sendKey = wlr_input_method_keyboard_grab_v2_send_key;
+
+        extern fn wlr_input_method_keyboard_grab_v2_send_modifiers(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, modifiers: *wlr.Keyboard.Modifiers) void;
+        pub const sendModifiers = wlr_input_method_keyboard_grab_v2_send_modifiers;
+
+        extern fn wlr_input_method_keyboard_grab_v2_set_keyboard(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, keyboard: *wlr.Keyboard) void;
```
Comment (ifreund) [inline src/types/input_method_v2.zig:60]:
```suggestion
        extern fn wlr_input_method_keyboard_grab_v2_set_keyboard(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab, keyboard: ?*wlr.Keyboard) void;
```

File: src/types/input_method_v2.zig [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        pub const setKeyboard = wlr_input_method_keyboard_grab_v2_set_keyboard;
+
+        extern fn wlr_input_method_keyboard_grab_v2_destroy(keyboard_grab: *wlr.InputMethodV2.KeyboardGrab) void;
+        pub const destroy = wlr_input_method_keyboard_grab_v2_destroy;
+    };
+
+    ressource: *wl.Ressource,
```
Comment (ifreund) [inline src/types/input_method_v2.zig:67]:
```suggestion
    resource: *wl.Resource,
```

File: src/types/input_method_v2.zig [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    seat: *wlr.Seat,
+    seat_client: *wlr.Seat.Client,
+
+    pending: wlr.InputMethodV2.State,
+    current: wlr.InputMethodV2.State,
+    active: bool,
+    client_active: bool,
+    current_serial: u32,
+
+    link: wl.list.Link,
```
Comment (ifreund) [inline src/types/input_method_v2.zig:77]: You're missing the `keyboard_grab` field before this one.

File: src/types/input_method_v2.zig (hunk 5 above)
Comment (ifreund) [inline src/types/input_method_v2.zig:67]:
```suggestion
    resource: *wl.Resource,
```

**Review by ifreund — CHANGES_REQUESTED** [review]

File: src/types/input_method_v2.zig (hunk 6 above)
Comment (novakne) [reply src/types/input_method_v2.zig:77]: so `keyboard_grab: KeyboardGrab,`? 
Would it be better if I don't put `pub const KeyboardGrab` inside `InputMethodV2` struct

File: src/types/input_method_v2.zig (hunk 5 above)
Comment (novakne) [reply src/types/input_method_v2.zig:67]: Damn I'm sorry I write it in french everywhere

File: src/types/input_method_v2.zig (hunk 6 above)
Comment (ifreund) [reply src/types/input_method_v2.zig:77]: > so `keyboard_grab: KeyboardGrab,`?

yep, exactly.

> Would it be better if I don't put `pub const KeyboardGrab` inside `InputMethodV2` struct

How you have it now is fine I think.

**Review by ifreund — APPROVED** [review]
LGTM, thanks!

**Pull request merged**